components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/main/java/org/wso2/carbon/analytics/spark/core/internal/SparkAnalyticsExecutor.java

```diff
@@ -780,13 +781,19 @@ private String replaceTableNamesInQuery(int tenantId, String query) {
         }
     }
 
-    private static AnalyticsQueryResult toResult(DataFrame dataFrame)
+    private AnalyticsQueryResult toResult(DataFrame dataFrame)
             throws AnalyticsExecutionException {
-        return new AnalyticsQueryResult(dataFrame.schema().fieldNames(),
-                                        convertRowsToObjects(dataFrame.collect()));
+        int resultsLimit = this.sparkConf.getInt("carbon.spark.results.limit", -1);
+        if(resultsLimit != -1)
```

Comment:
Fix code formatting (indentation), and put curly braces for even single line statements.